### PR TITLE
Update dm-release-notes.md

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-release-notes.md [hunk 1]
```diff
@@ -4,7 +4,10 @@ title: TiDB Data Migration 版本发布历史
 
 # TiDB Data Migration 版本发布历史
 
-TiDB Data Migration v5.3.0 及之前历史版本发布声明如下：
+从 DM v5.4 起，TiDB Data Migration 的 Release Notes 已合入相同版本号的 TiDB Release Notes。
```
Comment (shichun-0415) [inline dm/dm-release-notes.md:7]:
```suggestion
从 DM v5.4 起，TiDB Data Migration 的 Release Notes 合入相同版本号的 TiDB Release Notes。
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

File: dm/dm-release-notes.md [hunk 2]
```diff
@@ -4,7 +4,10 @@ title: TiDB Data Migration 版本发布历史
 
 # TiDB Data Migration 版本发布历史
 
-TiDB Data Migration v5.3.0 及之前历史版本发布声明如下：
+从 DM v5.4 起，TiDB Data Migration 的 Release Notes 合入相同版本号的 TiDB Release Notes。
+
+- 如需阅读 v5.4 及之后版本的 DM Release Notes，请查看对应版本的 TiDB Release Notes 中 DM 相关的内容。
```
Comment (qiancai) [inline dm/dm-release-notes.md:9]:
```suggestion
从 DM v5.4.0 起，TiDB Data Migration 的 Release Notes 合入相同版本号的 TiDB Release Notes。

- 如需阅读 v5.4.0 及之后版本的 DM Release Notes，请查看对应版本的 TiDB Release Notes 中 DM 相关的内容。
```

**Pull request merged**